src/System.Management.Automation/engine/PSVersionInfo.cs

```diff
@@ -544,24 +544,24 @@ private static bool TryParseVersion(string version, ref VersionResult result)
 
             var versionSansLabel = (dashIndex < 0) ? version : version.Substring(0, dashIndex);
             string[] parsedComponents = versionSansLabel.Split(Utils.Separators.Dot);
-            if (parsedComponents.Length != 3)
+            if (parsedComponents.Length == 0 || parsedComponents.Length > 3)
```

Comment:
So it's a one element array that contains an empty string.  In any case, there is already code that causes an empty string to fail before the split() statement.